### PR TITLE
Prefix `DeviceBuffer`'s C functions

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/rapidsai/rmm/issues/178

To avoid conflicts between Python only and C only methods, prefix C only methods with `c_`.

cc @kkraus14 @shwina

### CODE REVIEW

**Comment by jakirkham** [comment]
Just as a note, PR ( https://github.com/rapidsai/cudf/pull/3254 ) may need to be updated if we go through with these changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 - PR #106 Added multi-GPU initialization
 - PR #167 Added value setter to `device_scalar`
 - PR #163 Add Cython bindings to `device_buffer`
-- PR #177 Add __cuda_array_interface__ to DeviceBuffer
+- PR #177 Add `__cuda_array_interface__` to `DeviceBuffer`
```
Comment (kkraus14) [inline CHANGELOG.md]: Thanks :)

Comment (jakirkham) [reply CHANGELOG.md]: Of course 😄

**Comment by jakirkham** [comment]
Should add left alone `cpdef` functions as there is no risk of those conflicting between Python and C implementations as they are the same. 😉

**Pull request merged**